src/families/cosmos/types.js

```diff
@@ -0,0 +1,140 @@
+// @flow
+
+import type { BigNumber } from "bignumber.js";
+import type {
+  TransactionCommon,
+  TransactionCommonRaw
+} from "../../types/transaction";
+
+export type CoreStatics = {};
+export type CoreAccountSpecifics = {};
+export type CoreOperationSpecifics = {};
+export type CoreCurrencySpecifics = {};
+
+export type CosmosDelegation = {|
+  validatorAddress: string,
+  amount: BigNumber,
+  pendingRewards: BigNumber,
+  isActive: boolean, // TBD is there more status case? if so, switch to status field with an enum string.
+|};
+
+export type CosmosRedelegation = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosUnbounding = {|
+  validatorAddress: string,
+  completionTime: Date,
+  amount: BigNumber,
+|};
+
+export type CosmosResources = {|
+  delegations: CosmosDelegation[],
+  redelegations: CosmosRedelegation[],
+  unbounding: CosmosUnbounding[],
+  delegatedBalance: BigNumber,
+  pendingRewardsBalance: BigNumber, 
+  unboundingBalance: BigNumber,
+  withdrawAddress: string,
+|};
+
+export type CosmosDelegationRaw = {|
+  validatorAddress: string,
+  amount: string,
+  pendingRewards: string,
+  isActive: boolean,
+|};
+
+export type CosmosRedelegationRaw = {|
+  sourceValidatorAddress: string,
+  dstValidatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosUnboundingRaw = {|
+  validatorAddress: string,
+  completionTime: string,
+  amount: string,
+|};
+
+export type CosmosResourcesRaw = {|
+  delegations: CosmosDelegationRaw[],
+  redelegations: CosmosRedelegationRaw[],
+  unbounding: CosmosUnboundingRaw[],
+  delegatedBalance: string,
+  pendingRewardsBalance: string, 
+  unboundingBalance: string,
+  withdrawAddress: string,
+|};
+
+// NB this must be serializable (no Date, no BigNumber)
+export type CosmosValidatorItem = {|
+  validatorAddress: string,
+  name: string,
+  votingPower: number, // value from 0.0 to 1.0 (normalized percentage)
+  commission: number, // value from 0.0 to 1.0 (normalized percentage)
+|};
+
+// by convention preload would return a Promise of CosmosPreloadData
+export type CosmosPreloadData = {
+  validators: CosmosValidatorItem[]
+};
+
+export type CosmosOperationMode =
+  | "send"
+  | "delegate"
+  | "undelegate"
+  | "redelegate"
+  | "claimReward"
+  | "claimAllRewards";
+
+export type NetworkInfo = {|
+  family: "cosmos",
+  fees: BigNumber
+|};
+
+export type NetworkInfoRaw = {|
+  family: "cosmos",
+  fees: string
+|};
+
+// TODO suggest to rename to CosmosValidatorInput?
+export type CosmosValidator = {
+  address: string,
+  amount: BigNumber
```

Comment:
What is this amount supposed to represent ?

Reply:
this was part of the existing stuff. cc @henri-ly 
this is user's input, how much he set to vote. that's why i suggested to rename CosmosValidatorInput

IDK a good wording btw, shouldn't it be like "CosmosVote" ?

Reply:
Oh ok, then I think the amount is the voting power of the validator, because that information is already in `CosmosDelegation` (At some point we'll have to decide if we follow the naming schemes of the coins, or our own naming scheme that we document somewhere. `TronVote` = `CosmosDelegation` and `CosmosVote` is kind of confusing because that's another feature that's uncovered here)

Reply:
It's what @gre said it's the user input of how much we are going to delegate and to who.
it works like this : 

> --cosmosValidator addr1, addr2 --cosmosAmountValidator 0.33, 1.322

wasn't really inspired for the name and if we can change to something clearer i'm in !

Reply:
what is the best wording for such "vote" that goes into a delegation transaction? curious about COSMOS' semantic here 🤔